android/src/main/java/com/dylanvann/fastimage/OkHttpProgressGlideModule.java

```diff
@@ -64,15 +65,12 @@ public static void expect(String key, ProgressListener listener) {
     }
 
     private interface ResponseProgressListener {
-
         void update(String key, long bytesRead, long contentLength);
     }
 
     private static class DispatchingProgressListener implements ResponseProgressListener {
-
-        private static final Map<String, ProgressListener> LISTENERS = new WeakHashMap<>();
-
-        private static final Map<String, Long> PROGRESSES = new WeakHashMap<>();
+        private static final Map<String, ProgressListener> LISTENERS = new HashMap<>();
```

Comment:
Did'nt we leak the ProgressListener class here ?

Reply:
Should not be static, isn't it ?

Reply:
```java
private final Map<String, ProgressListener> LISTENERS = new WeakHashMap<>();
private final Map<String, Long> PROGRESSES = new HashMap<>();
```

Reply:
Same, issue as the other comment, removing static for next build.

Reply:
I reckon there is another method below that removes the instances form the Map, therefor is it actually removed form memory and no leaked?

 static void forget(String key) {
            LISTENERS.remove(key);
            PROGRESSES.remove(key);
        }

Reply:
Switching Hashmap for HashMapWeak